django/db/backends/sqlite3/base.py

```diff
@@ -264,37 +265,68 @@ def check_constraints(self, table_names=None):
         determine if rows with invalid references were entered while constraint
         checks were off.
         """
-        with self.cursor() as cursor:
-            if table_names is None:
-                table_names = self.introspection.table_names(cursor)
-            for table_name in table_names:
-                primary_key_column_name = self.introspection.get_primary_key_column(cursor, table_name)
-                if not primary_key_column_name:
-                    continue
-                key_columns = self.introspection.get_key_columns(cursor, table_name)
-                for column_name, referenced_table_name, referenced_column_name in key_columns:
-                    cursor.execute(
-                        """
-                        SELECT REFERRING.`%s`, REFERRING.`%s` FROM `%s` as REFERRING
-                        LEFT JOIN `%s` as REFERRED
-                        ON (REFERRING.`%s` = REFERRED.`%s`)
-                        WHERE REFERRING.`%s` IS NOT NULL AND REFERRED.`%s` IS NULL
-                        """
-                        % (
-                            primary_key_column_name, column_name, table_name,
-                            referenced_table_name, column_name, referenced_column_name,
-                            column_name, referenced_column_name,
+        if Database.sqlite_version_info >= (3, 20, 0):
```

Comment:
@carltongibson support for `PRAGMA foreign_key` was only added in SQLite 3.20.

https://www.sqlite.org/releaselog/3_20_0.html

Before that it would silently execute and never return any row which is the default behaviour for unsupported pragmas.

Reply:
OK. Are we (i.e. are you 🙂) happy with the magic number here? (There'll be `git blame` to this commit and discussion so it's arguably enough.)

Reply:
I am, we usually only use these version gates to deal mark support or not in `django.db.backends`.

Reply:
OK, done. 🙂

Reply:
I think we've generally tried to make all comparisons like this as database features. A feature name improves readability and puts all version references in one place for easier removal when dropping support for old versions.